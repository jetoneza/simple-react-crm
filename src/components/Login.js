import React from 'react';
import Validator from 'validator';
import { Form } from 'semantic-ui-react';

class Login extends React.Component {
  state = {
    data: {
      email: '',
      password: '',
    },
    errors: {},
  }

  handleSubmit = (e) => {
    e.preventDefault();

    const errors = this.validate(this.state.data);

    this.setState({ errors });

    if (Object.keys(errors).length !== 0) {
      return;
    }

    // TODO: handle submit
    this.props.history.push('/');
  }

  validate = (data) => {
    const errors = {};

    if (!data.password) {
      errors.password = 'Please provide a password.';
    }

    if (!Validator.isEmail(data.email)) {
      errors.email = 'Invalid email.';
    }

    return errors;
  }

  handleOnChange = (e) => {
    const { name, value } = e.target;

    this.setState({
      data: {
        ...this.state.data,
        [name]: value,
      },
    })
  }

  renderError = (error) => {
    return error && <span style={{ color: 'red' }}>{ error }</span>;
  }

  render() {
    const { data, errors } = this.state;

    return (
      <div className="ui middle aligned center aligned grid">
        <div className="column" style={{ maxWidth: 450 }}>
          <h2 className="ui green image header">
            <div className="content">
              Log-in to your account
            </div>
          </h2>
          <Form onSubmit={this.handleSubmit}>
            <div className="ui stacked segment">
              <Form.Field error={!!errors.email}>
                <div className="ui left icon input">
                  <i className="user icon"></i>
                  <input
                    type="email"
                    name="email"
                    placeholder="E-mail address"
                    onChange={this.handleOnChange}
                    value={data.email}/>
                </div>
                { this.renderError(errors.email) }
              </Form.Field>
              <Form.Field error={!!errors.password}>
                <div className="ui left icon input">
                  <i className="lock icon"></i>
                  <input
                    type="password"
                    name="password"
                    placeholder="Password"
                    onChange={this.handleOnChange}
                    value={data.password}/>
                </div>
                { this.renderError(errors.password) }
              </Form.Field>
              <button className="ui fluid large green submit button">Login</button>
            </div>
          </Form>
        </div>
      </div>
    );
  }
}

export default Login;
