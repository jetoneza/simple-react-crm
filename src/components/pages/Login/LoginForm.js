import React from 'react';
import PropTypes from 'prop-types';
import Validator from 'validator';
import { Form } from 'semantic-ui-react';

class LoginForm extends React.Component {
  static propTypes = {
    onSubmit: PropTypes.func.isRequired,
  };

  state = {
    data: {
      email: '',
      password: '',
    },
    errors: {},
  };

  handleSubmit = (e) => {
    e.preventDefault();

    const errors = this.validate(this.state.data);

    this.setState({ errors });

    if (Object.keys(errors).length !== 0) {
      return;
    }

    this.props.onSubmit(this.state.data);
  };

  validate = (data) => {
    const errors = {};

    if (!data.password) {
      errors.password = 'Please provide a password.';
    }

    if (!Validator.isEmail(data.email)) {
      errors.email = 'Invalid email.';
    }

    return errors;
  };

  handleOnChange = (e) => {
    const { name, value } = e.target;

    this.setState({
      data: {
        ...this.state.data,
        [name]: value,
      },
    });
  };

  renderError = (error) => {
    return error && <span style={{ color: 'red' }}>{ error }</span>;
  };

  render() {
    const { data, errors } = this.state;

    return (
      <Form onSubmit={this.handleSubmit}>
        <div className="ui stacked segment">
          <Form.Field error={!!errors.email}>
            <div className="ui left icon input">
              <i className="user icon"></i>
              <input
                type="email"
                name="email"
                placeholder="E-mail address"
                onChange={this.handleOnChange}
                value={data.email}/>
            </div>
            { this.renderError(errors.email) }
          </Form.Field>
          <Form.Field error={!!errors.password}>
            <div className="ui left icon input">
              <i className="lock icon"></i>
              <input
                type="password"
                name="password"
                placeholder="Password"
                onChange={this.handleOnChange}
                value={data.password}/>
            </div>
            { this.renderError(errors.password) }
          </Form.Field>
          <button className="ui fluid large green submit button">Login</button>
        </div>
      </Form>
    );
  }
}

export default LoginForm;
