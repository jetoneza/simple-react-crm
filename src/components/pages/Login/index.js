import React from 'react';
import PropTypes from 'prop-types';

import LoginForm from './LoginForm';

class Login extends React.PureComponent {
  static propTypes = {
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }).isRequired,
  };

  handleOnSubmit = (data) => {
    // TODO: handle submit
    this.props.history.push('/');
  };

  render() {
    return (
      <div className="ui middle aligned center aligned grid">
        <div className="column" style={{ maxWidth: 450 }}>
          <h2 className="ui green image header">
            <div className="content">
              Log-in to your account
            </div>
          </h2>
          <LoginForm onSubmit={this.handleOnSubmit}/>
        </div>
      </div>
    );
  }
}

export default Login;
