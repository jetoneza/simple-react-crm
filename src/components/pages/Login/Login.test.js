import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import { shallow } from 'enzyme';

import Login from './index';
import LoginForm from './LoginForm';

describe('<Login /> test', () => {
  const pushCallback = sinon.spy();
  const props = {
    history: {
      push: pushCallback,
    },
  };

  test('Should render login form', () => {
    const wrapper = shallow(<Login {...props}/>);

    expect(wrapper.find(LoginForm).length).toEqual(1);
  });

  test('Should execute .handleOnSubmit correctly', () => {
    const wrapper = shallow(<Login {...props}/>);

    const handleOnSubmit = wrapper.instance().handleOnSubmit;

    expect(handleOnSubmit).not.toBe('undefined');

    handleOnSubmit(); // Simulate invoking handleOnSubmit

    expect(pushCallback.called).toBeTruthy();
  });

  test('Should render login page', () => {
    const component = <Login {...props}/>
    shallow(component);

    const renderedComponent = renderer.create(component);
    const tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
