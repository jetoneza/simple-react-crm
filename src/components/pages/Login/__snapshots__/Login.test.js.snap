// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> test Should render login page 1`] = `
<div
  className="ui middle aligned center aligned grid"
>
  <div
    className="column"
    style={
      Object {
        "maxWidth": 450,
      }
    }
  >
    <h2
      className="ui green image header"
    >
      <div
        className="content"
      >
        Log-in to your account
      </div>
    </h2>
    <form
      action={undefined}
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="ui stacked segment"
      >
        <div
          className="field"
        >
          <div
            className="ui left icon input"
          >
            <i
              className="user icon"
            />
            <input
              name="email"
              onChange={[Function]}
              placeholder="E-mail address"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui left icon input"
          >
            <i
              className="lock icon"
            />
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          className="ui fluid large green submit button"
        >
          Login
        </button>
      </div>
    </form>
  </div>
</div>
`;
